### PR TITLE
Add clippy to the tools list

### PR DESCRIPTION
**Pull request by yaahc**
Implements https://github.com/rust-lang-nursery/rust-clippy/issues/2901

### CODE REVIEW

File: src/rustup-cli/self_update.rs [hunk 1]
```diff
@@ -190,7 +190,7 @@ tools, but otherwise, install the C++ build tools before proceeding.
 "#;
 
 static TOOLS: &'static [&'static str] =
-    &["rustc", "rustdoc", "cargo", "rust-lldb", "rust-gdb", "rls"];
+    &["rustc", "rustdoc", "cargo", "rust-lldb", "rust-gdb", "rls", "clippy-driver", "cargo-clippy"];
```
Comment (oli-obk) [inline src/rustup-cli/self_update.rs]: So if I remember the discussion on discord correctly, we don't actually need `clippy-driver` here, right?

File: src/rustup-win-installer/src/lib.rs [hunk 2]
```diff
@@ -26,6 +26,8 @@ static TOOLS: &'static [&'static str] = &[
     "rls",
     "rustfmt",
     "cargo-fmt",
+    "clippy-driver",
```
Comment (oli-obk) [inline src/rustup-win-installer/src/lib.rs]: same here

**Comment by yaahc** [comment]
forgot to push before leaving for the weekend. Should be good now

**Comment by alexcrichton** [comment]
Thanks! Can this also update `DUP_TOOLS` below the definition of `TOOLS`? That'll help encourage users to uninstall the `cargo install`'d version, if it exists, before adding the rustup component

**Comment by oli-obk** [comment]
@alexcrichton While that made sense for rustfmt (because there was stable rustfmt), for clippy it doesn't make much sense. If you turn on the component, your manually installed clippy was probably already broken and will never work again.

I think we should just silently overwrite an existing installation

**Comment by alexcrichton** [comment]
Ah yes, an excellent point!

**Comment by alexcrichton** [comment]
For those following along, this is now published as rustup 1.13.0 and should be good to go!

**Pull request merged**